--- ./kernel/nvidia/nvidia.Kbuild
+++ ./kernel/nvidia/nvidia.Kbuild
@@ -203,3 +203,4 @@ NV_CONFTEST_GENERIC_COMPILE_TESTS += get_user_pages_remote
 NV_CONFTEST_GENERIC_COMPILE_TESTS += pm_runtime_available
 NV_CONFTEST_GENERIC_COMPILE_TESTS += vm_fault_t
 NV_CONFTEST_GENERIC_COMPILE_TESTS += pci_class_multimedia_hd_audio
+ldflags-y += -T /opt/vgpu_unlock/kern.ld

--- ./kernel/nvidia/os-interface.c
+++ ./kernel/nvidia/os-interface.c
@@ -16,7 +16,7 @@

 #include "nv-time.h"

-
+#include "/opt/vgpu_unlock/vgpu_unlock_hooks.c"

